FSI/flap_perp_2D/SU2-FEniCS/Solid/perp-flap.py

```diff
@@ -0,0 +1,249 @@
+    if ufl:
+        dt_ = dt
+        beta_ = beta
+    else:
+        dt_ = float(dt)
+        beta_ = float(beta)
+
+    return ((u - u_old - dt_ * v_old) / beta / dt_ ** 2
+            - (1 - 2 * beta_) / 2 / beta_ * a_old)
+
+
+# Update velocity
+def update_v(a, u_old, v_old, a_old, ufl=True):
+    if ufl:
+        dt_ = dt
+        gamma_ = gamma
+    else:
+        dt_ = float(dt)
+        gamma_ = float(gamma)
+
+    return v_old + dt_ * ((1 - gamma_) * a_old + gamma_ * a)
+
+
+def update_fields(u, u_old, v_old, a_old):
+    """Update all fields at the end of a timestep."""
+
+    u_vec, u0_vec = u.vector(), u_old.vector()
+    v0_vec, a0_vec = v_old.vector(), a_old.vector()
+
+    # call update functions
+    a_vec = update_a(u_vec, u0_vec, v0_vec, a0_vec, ufl=False)
+    v_vec = update_v(a_vec, u0_vec, v0_vec, a0_vec, ufl=False)
+
+    # assign u->u_old
+    v_old.vector()[:], a_old.vector()[:] = v_vec, a_vec
+    u_old.vector()[:] = u.vector()
+
+
+def avg(x_old, x_new, alpha):
+    return alpha * x_old + (1 - alpha) * x_new
+
+
+# residual
+a_np1 = update_a(du, u_n, v_n, a_n, ufl=True)
+v_np1 = update_v(a_np1, u_n, v_n, a_n, ufl=True)
+
+res = m(avg(a_n, a_np1, alpha_m), v) + k(avg(u_n, du, alpha_f), v)
+
+a_form = lhs(res)
+L_form = rhs(res)
+
+# parameters for Time-Stepping
+t = 0.0
+n = 0
+time = []
+u_tip = []
+time.append(0.0)
+u_tip.append(0.0)
+E_ext = 0
+
+displacement_out = File("Solid/FSI-S/u_fsi.pvd")
+
+u_n.rename("Displacement", "")
+u_np1.rename("Displacement", "")
+displacement_out << u_n
+
+while precice.is_coupling_ongoing():
+
+    if precice.is_action_required(precice.action_write_iteration_checkpoint()):  # write checkpoint
+        precice.store_checkpoint(u_n, t, n)
+        u_ref = u_n.copy()
+
+    # read data from preCICE and get a new coupling expression
+    read_data = precice.read_data()
+
+    # Update the point sources on the coupling boundary with the new read data
+    Forces_x, Forces_y = precice.get_point_sources(read_data)
+
+    A, b = assemble_system(a_form, L_form, bc)
+
+    b_forces = b.copy()  # b is the same for every iteration, only forces change
+
+    for ps in Forces_x:
+        ps.apply(b_forces)
+    for ps in Forces_y:
+        ps.apply(b_forces)
+
+    assert (b is not b_forces)
+    solve(A, u_np1.vector(), b_forces)
+
+    dt = Constant(np.min([precice_dt, fenics_dt]))
+
+    # Write new displacements to preCICE
+    u_delta = project(u_np1 - u_ref, V)
+    precice.write_data(u_delta)
+
+    # Call to advance coupling, also returns the optimum time step value
+    precice_dt = precice.advance(dt(0))
+
+    # Either revert to old step if timestep has not converged or move to next timestep
+    if precice.is_action_required(precice.action_read_iteration_checkpoint()):  # roll back to checkpoint
+        u_cp, t_cp, n_cp = precice.retrieve_checkpoint()
+        u_n.assign(u_cp)
+        t = t_cp
+        n = n_cp
+    else:
+        u_n.assign(u_np1)
+        t += dt
```

Comment:
```suggestion
        t += float(dt)
```
See https://github.com/precice/tutorials/pull/115